src/output.rs

```diff
@@ -0,0 +1,82 @@
+// commands.rs
+
+// Copyright (C) 2020 The Nitrocli Developers
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+use crate::Context;
+
+/// A progress bar that can be printed to an interactive output.
+pub struct ProgressBar {
+  redraw: bool,
+  progress: u8,
+  toggle: bool,
+  finished: bool,
+}
+
+impl ProgressBar {
+  pub fn new() -> ProgressBar {
+    ProgressBar {
+      redraw: true,
+      progress: 0,
+      toggle: false,
+      finished: false,
+    }
+  }
+
+  pub fn is_finished(&self) -> bool {
+    self.finished
+  }
+
+  pub fn update(&mut self, progress: u8) -> anyhow::Result<()> {
+    anyhow::ensure!(!self.finished, "Tried to update finished progress bar");
+    anyhow::ensure!(
+      progress <= 100,
+      "Progress bar value out of range: {}",
+      progress
+    );
+    if progress != self.progress {
+      self.redraw = true;
+      self.progress = progress;
+    }
+    self.toggle = !self.toggle;
+    Ok(())
+  }
+
+  pub fn finish(&mut self) {
+    self.finished = true;
+    self.redraw = true;
+    self.progress = 100;
+  }
+
+  pub fn draw(&self, ctx: &mut Context<'_>) -> anyhow::Result<()> {
+    use crossterm::{cursor, terminal};
```

Comment:
Oh...guess we need it for other stuff too :-|

Have you considered using [`termion`](https://crates.io/crates/termion)? Would it fulfill our requirements?

Reply:
Yes. `termion` works for ANSI terminals only.  `crossterm` is cross-platform and also supports e. g. Windows terminals.